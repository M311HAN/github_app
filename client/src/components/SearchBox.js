import React, { useState, useEffect } from "react";

const SearchBox = ({ onSearch }) => {
  // State to manage the input value
  const [username, setUsername] = useState("");

  useEffect(() => {
    // Clear input when the component mounts or certain conditions are met
    setUsername("");
  }, []);

  const handleSubmit = (event) => {
    // Prevent the default form submission behavior
    event.preventDefault();
    // Trigger the search with the current username
    onSearch(username);
  };

  return (
    <div className="d-flex justify-content-center mt-5">
      <div
        className="card p-4 shadow"
        style={{
          width: "50%",
          backgroundColor: "#343a40",
          borderRadius: "10px",
        }}
      >
        <form onSubmit={handleSubmit} className="d-flex">
          <input
            type="text"
            className="form-control me-2"
            placeholder="Search GitHub User..."
            value={username}
            onChange={(event) => setUsername(event.target.value)}
            style={{ borderRadius: "5px", padding: "10px" }}
          />
          <button
            type="submit"
            className="btn btn-primary"
            style={{ padding: "10px 20px", borderRadius: "5px" }}
          >
            Search
          </button>
        </form>
      </div>
    </div>
  );
};

export default SearchBox;
