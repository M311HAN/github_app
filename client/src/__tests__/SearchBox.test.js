import React from "react";
import { render } from "@testing-library/react";
// Import the SearchBox component to be tested
import SearchBox from "../components/SearchBox";

test("renders correctly", () => {
  // Render the SearchBox component with a dummy onSearch prop
  const { asFragment } = render(<SearchBox onSearch={() => {}} />);
  // Check that the rendered component matches the previous snapshot
  expect(asFragment()).toMatchSnapshot();
});
