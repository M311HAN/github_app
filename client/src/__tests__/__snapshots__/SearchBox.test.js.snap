// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center mt-5"
  >
    <div
      class="card p-4 shadow"
      style="width: 50%; background-color: rgb(52, 58, 64); border-radius: 10px;"
    >
      <form
        class="d-flex"
      >
        <input
          class="form-control me-2"
          placeholder="Search GitHub User..."
          style="border-radius: 5px; padding: 10px;"
          type="text"
          value=""
        />
        <button
          class="btn btn-primary"
          style="padding: 10px 20px; border-radius: 5px;"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
